src/core/SampleBuffer.cpp

```diff
@@ -1447,3 +1457,167 @@ SampleBuffer::handleState::~handleState()
 {
 	src_delete( m_resamplingData );
 }
+
+
+
+// Fading related functions
+
+float SampleBuffer::fadingVal(f_cnt_t pos, bool is_left)
+{
+    float relativePos = static_cast<float>(pos)/static_cast<float>(m_frames);
+    return (this->*fadefuncPtr)(relativePos, is_left);
+}
+
+
+void SampleBuffer::setLeftFader(float lfader)
+{
+    m_leftFader = lfader;
+    checkFadingActive();
+}
+
+
+void SampleBuffer::setRightFader(float rfader)
+{
+    m_rightFader = rfader;
+    checkFadingActive();
+}
+
+
+void SampleBuffer::checkFadingActive()
+{
+    if ((m_leftFader > 0.0) || (m_rightFader < 1.0))
+    {
+        m_fading = true;
+    }
+    else
+    {
+        m_fading = false;
+    }
+}
+
+
+float SampleBuffer::bezierFade(float relativePos, bool is_left)
+{
+    float amp = 1.0f;
+    if (is_left)
+    {
+        amp *= m_leftBezierFade.evaluate(relativePos/m_leftFader);
+    }
+    else
+    {
+        amp *= m_rightBezierFade.evaluate(1-(relativePos-m_rightFader)/(1-m_rightFader));
+    }
+    return amp;
+}
+
+
+float CubicBezier::cubicBezierLastSegmentP1(float t, float p1, float p2)
+{
+    return 2*p1*t*(1-t) + p2*pow(t, 2);
```

Comment:
```suggestion
    return 2 * p1 * t * (1 - t) + p2 * pow(t, 2);
```

I'm not meaning to nitpick, tbh on the context of the Bezier formula I even think the way you written it might be more readable. But our code style guidelines says we need spaces around operators. I'm wondering if there are/should be exceptions. This repeats on other lines but I'm not marking them all because I want to hear from others if we should open exceptions where the formula is more readable without spaces on every operator (like here).

Reply:
So bringing it up on the discord chat an alternative has been brought up by @Veratil which can make the polynomial equation more readable while still conforming to our code style:
```
// Term with power of 1
auto e1 = 2 * p1 * t * (1 - t);
// Term with power of 2
auto e2 = p2 * pow(t, 2);
// Polynomial equation
return e1 + e2;
```

Compiler optimization makes sure the code is compiled to the same as if we had no variables holding the equation terms.

Another alternative that is simpler and might be good enough here is:
```
return (p2) * pow(t, 2) + (2 * p1 * (1 - t)) * t;
```
Which just puts the coefficients on parenthesis and leave the independent variable outside. Just some ideas, you can just add the spaces if you think it's good enough.